libtpms: 0.7.4 -> 0.8.0

baloo opened the pull request:
###### Motivation for this change

Bump version to 0.8.0 and adds tpm2 support by default.

supplement #114479

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

r-ryantm committed:
libtpms: 0.7.4 -> 0.8.0

baloo committed:
libtpms: tpm2 support is out of experimental

Since db80bd9ea16894a1902c3ab787aea9d58e7d1e85 commit, tpm2 support is
not experimental anymore

Signed-off-by: Arthur Gautier <baloo@superbaloo.net>

r-rmcgibbo commented:
Result of `nixpkgs-review pr 114491` at 2a580f7f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/a1cc327d4afa07eae313ffc1cb600007">libtpms</a></li></li>
    <li><a href="https://gist.github.com/26218caaa20bde8903979637ba2d572c">swtpm</a></li> (swtpm-tpm2)</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/tools/security/libtpms/default.nix:
```diff
@@ -3,18 +3,18 @@
 , pkg-config, autoreconfHook
 , openssl, perl
-, tpm2Support ? false
+, tpm2Support ? true
```
Can we remove this flag?

baloo replied:
I guess we can.

Pamplemousse commented:
Result of `nixpkgs-review pr 114491` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>libtpms</li>
    <li>swtpm (swtpm-tpm2)</li>
  </ul>
</details>

Pamplemousse reviewed (approved):
LGTM :+1:
Dependent `swtpm` binaries seems to be running ok (with `--help`).

baloo commented:
@SuperSandro2000 any update on this?

baloo committed:
libtpms: remove extraneous output

Nothing was put in the $out output, remove the $lib and put everything
that was in it in the $out.

Signed-off-by: Arthur Gautier <baloo@superbaloo.net>

SuperSandro2000 commented:
@ofborg eval

tomberek commented:
Result of `nixpkgs-review pr 114491` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>libtpms</li>
    <li>swtpm (swtpm-tpm2)</li>
  </ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114491` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>libtpms</li>
<li>swtpm (swtpm-tpm2)</li>
</ul>
</details>
